tensorflow/python/framework/sparse_tensor.py

```diff
@@ -178,6 +178,21 @@ def values(self):
     """
     return self._values
 
+  def with_values(self, new_values):
+    """Returns a copy of `self` with `values` replaced by `new_values`.
+
+    This method produces a new `SparseTensor` that has the same nonzero
+    indices, but updated values.
+
+    Args:
+      new_values: The values of the new `SparseTensor. Needs to have the same
+        shape as the current `.values` `Tensor`.
+
+    Returns:
+      A `SparseTensor` with identical indices but updated values.
```

Comment:
Might be worth explicitly mentioning shape here (and above) too.  I.e.:

...with identical indices and shape...

or:

...with identical indices and dense shape...

Reply:
should be done with the latest commig